src/common/containers/Profile.js

```diff
@@ -1,28 +1,102 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { object, func } from 'prop-types';
+import { withRouter } from 'react-router-dom';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
-import { withRouter } from 'react-router';
+import { withStyles } from '@material-ui/core/styles';
 import Paper from '@material-ui/core/Paper';
 
+import RecentCardsCarousel from '../components/cards/RecentCardsCarousel';
+import Title from '../components/Title';
+import * as collectionActions from '../actions/collection';
+
 export function mapStateToProps(state) {
   return {
-    loggedIn: state.global.user !== null
+    user: state.global.user
   };
 }
 
-const About = function (props) {
-  return (
-    <div style={{margin: '48px 72px'}}>
-      <Helmet title="Profile"/>
+export function mapDispatchToProps(dispatch) {
+  return {
+    onOpenForEditing: (card) => {
+      dispatch(collectionActions.openForEditing(card));
+    }
+  };
+}
 
-      <div style={{display: 'flex', justifyContent: 'stretch'}}>
-        <Paper style={{padding: '5px 20px'}}>
-          <h2>Profile</h2>
-          <p>TODO: Stuff goes here.</p>
-        </Paper>
-      </div>
-    </div>
-  );
+const styles = {
+  container: {
+    margin: 20
+  },
+  paperContainer: {
+    display: 'flex',
+    justifyContent: 'space-between'
+  },
+  gridItem: {
+    width: 'calc((100% / 3) - (40px / 3))',
+    height: 600
+  },
+  recentCardsContainer: {
+    marginTop: 20,
+    padding: '5px 25px'
+  }
 };
 
-export default withRouter(connect(mapStateToProps)(About));
+class Profile extends Component {
+  constructor(props) {
+    super(props);
+
+    if (!this.userFromPath) {
+      props.history.push('/');
+    }
+
+    this.state = {
+      userId: this.userFromPath
+    };
+  }
+
+  static propTypes = {
+    user: object,
+    classes: object,
+    history: object,
+
+    onOpenForEditing: func
+  }
+
+  get userFromPath() {
```

Comment:
Does this logic need to live here? Can't you put it on the `Profile` route in `App`?
Replace
```
<Route path="/profile" component={Profile} />
```
with something like:
```
<Route path="/profile/:userId" component={Profile} />
```
and then within `Profile` you can use `props.match.params.userId`.

Reply:
Ooh thats much more elegant